dlc-manager/src/manager.rs

```diff
@@ -470,47 +472,59 @@ where
         Err(e)
     }
 
-    fn check_signed_contract(&mut self, contract: &SignedContract) -> Result<(), Error> {
+    fn check_signed_contract(
+        &mut self,
+        contract: &SignedContract,
+    ) -> Result<Option<ContractId>, Error> {
```

Comment:
It feels like it'd be easier to just return a boolean here, indicating whether the contract was updated or not.

Reply:
smart!!!